### DISCUSSION TITLE
Pass LIKE operator inside of IN operator

### DISCUSSION THREAD

**Discussion by sumchans** [supabase/supabase#2938] [Questions] [answered]
I am trying to get answers to this from somewhere. I have posted the same to a couple different forums already, not able to get a solution so far. Below is a query copied from the supabase website.
```
const { data, error } = await supabase
  .from('users')
  .select('phone_number')
  .in('phone_number', LIKE ANY(ARRAY[@contacts]))
```

I want to achieve the same but I need to pass a LIKE operator inside of the IN operator with a variable created in Dart. Something like below, which I know is not right.
```
const { data, error } = await supabase
  .from('users')
  .select('phone_number')
  .in('phone_number', LIKE ANY(ARRAY[@contacts]))
```
The contacts variable is of String type in dart and it has a format in this way 
`'%91960000000’,’%7780000000',’%6720000000',’%4160000000','%7780000000'`

**Comment by soedirgo** [comment, marked as answer]
I don't think you can use `LIKE ANY` with `.in()`. One way to do this is by using a function:
```sql
CREATE FUNCTION users_with_matching_phone_number(contacts text[]) RETURNS TABLE(phone_number text) as $$
SELECT * FROM users WHERE phone_number LIKE ANY(contacts);
$$ language sql;
```
```js
const contacts = ['%91960000000', '%7780000000', '%6720000000', '%4160000000', '%7780000000']
const { data, error } = await supabase
  .rpc('users_with_matching_phone_number', { contacts })
```

**Reply by sumchans** [reply]
Thanks @soedirgo, really appreciate taking the time to help out. I am almost there with your code. Just wondering if there is an issue with dart code not taking in paramaters? I get an error under the variable like below.

List<String> phoneContacts = [];
await queries.findContactsOnBol(phoneContacts);  
<img width="705" alt="Screen Shot 2021-08-23 at 8 58 44 PM" src="https://user-images.githubusercontent.com/33538243/130553685-373d9b1d-983c-4af4-becf-88ace2171db0.png">

**Reply by soedirgo** [reply]
I'm not familiar with Dart, I guess you can try `{ contacts: contacts }`?

**Reply by sumchans** [reply]
tried that as well, still I have the red line underneath.

**Reply by soedirgo** [reply]
@phamhieu any idea how to resolve this?

**Reply by silentworks** [reply]
@sumchans try

```js
const contacts = ['%91960000000', '%7780000000', '%6720000000', '%4160000000', '%7780000000']
const { data, error } = await supabase
  .rpc('users_with_matching_phone_number', params: { 'contacts': contacts })
```

**Reply by sumchans** [reply]
Thanks @soedirgo for changing the direction of my thinking on this issue. I never would have thought about creating a function to fix this issue.
Thanks @silentworks for putting a completion to it.

I just added an execute() method for the above query to return data -
```
const contacts = ['%91960000000', '%7780000000', '%6720000000', '%4160000000', '%7780000000']
const { data, error } = await supabase
  .rpc('users_with_matching_phone_number', params: { 'contacts': contacts }).execute();
```

This one works as expected.
